### PR TITLE
[NEXUS-4594] Skip adding retested items to NFC in case that repository is blocked

### PR DESCRIPTION
**Pull request by adreghiciu**
[NEXUS-4594] Skip adding retested items to NFC in case that repository is auto blocked or manually blocked

Signed-off-by: Alin Dreghiciu adreghiciu@gmail.com

### CODE REVIEW

File: nexus/nexus-proxy/src/main/java/org/sonatype/nexus/proxy/repository/AbstractProxyRepository.java [hunk 1]
```diff
@@ -1671,4 +1671,27 @@ public abstract class AbstractProxyRepository
...
+    protected boolean shouldAddToNotFoundCache( final ResourceStoreRequest request )
+    {
+        boolean shouldAddToNFC = super.shouldAddToNotFoundCache( request );
+        if ( shouldAddToNFC )
+        {
+            shouldAddToNFC = getProxyMode() == null || getProxyMode().shouldProxy();
+            if ( !shouldAddToNFC && getLogger().isDebugEnabled() )
+            {
+                getLogger().debug(
+                    String.format( "Repository '%s' is in proxy mode '%s', not adding path '%s' to NFC",
```
Comment (cstamas) [inline nexus/nexus-proxy/src/main/java/org/sonatype/nexus/proxy/repository/AbstractProxyRepository.java]: Typo: repository is NOT in proxy mode... but better would be "Proxy repository %s is not allowed to issue remote requests..."

**Comment by cstamas** [comment]
+1

File: nexus/nexus-proxy/src/main/java/org/sonatype/nexus/proxy/repository/AbstractProxyRepository.java (hunk 1 above)
Comment (adreghiciu) [inline nexus/nexus-proxy/src/main/java/org/sonatype/nexus/proxy/repository/AbstractProxyRepository.java]: Fixed in 67583e94

File: nexus/nexus-test-harness/nexus-test-harness-its/src/test/java/org/sonatype/nexus/integrationtests/nexus4539/AutoBlockITSupport.java [hunk 2]
```diff
@@ -19,8 +19,8 @@
 package org.sonatype.nexus.integrationtests.nexus4539;
 
 import static org.hamcrest.MatcherAssert.assertThat;
-import static org.hamcrest.Matchers.equalTo;
-import static org.hamcrest.Matchers.notNullValue;
+import static org.hamcrest.Matchers.*;
+import static org.hamcrest.Matchers.not;
```
Comment (nabcos) [inline nexus/nexus-test-harness/nexus-test-harness-its/src/test/java/org/sonatype/nexus/integrationtests/nexus4539/AutoBlockITSupport.java:23]: weird imports, some lines below Matchers is imported directly as well

**Comment by nabcos** [comment]
+1

File: nexus/nexus-test-harness/nexus-test-harness-its/src/test/java/org/sonatype/nexus/integrationtests/nexus4539/AutoBlockITSupport.java (hunk 2 above)
Comment (adreghiciu) [inline nexus/nexus-test-harness/nexus-test-harness-its/src/test/java/org/sonatype/nexus/integrationtests/nexus4539/AutoBlockITSupport.java:23]: Something is again wrong with my idea settings. I will fix that.

Comment (adreghiciu) [inline nexus/nexus-test-harness/nexus-test-harness-its/src/test/java/org/sonatype/nexus/integrationtests/nexus4539/AutoBlockITSupport.java:23]: Fixed in be4e3fe3

**Pull request closed without merging**